math/binary_exponent.cpp

```diff
@@ -1,14 +1,21 @@
-/// C++ Program to find Binary Exponent recursively.
+/// C++ Program to find Binary Exponent Iteratively and Recursively.
 
 #include<iostream>
 /*
- * Calculate a^b in O(log(b)) by converting b to a binary number
- * Binary exponentiation (also known as exponentiation by squaring)
- * is a trick which allows to calculate an using only O(logn) multiplications
- * (instead of O(n) multiplications required by the naive approach).
+ * Calculate a^b in O(log(b)) by converting b to a binary number.
+ * Binary exponentiation is also known as exponentiation by squaring.
+ * NOTE : This is a far better approach compared to naive method which provide O(b) operations.
+ * Example:
+ * 10 in base 2 is 1010.
+ * 2^10 = 2^(1010) = 2^8 * 2^2
+ * 2^1 = 2
+ * 2^2 = (2^1)^2 = 2^2 = 4
+ * 2^4 = (2^2)^2 = 4^2 = 16
+ * 2^8 = (2^4)^2 = 16^2 = 256
+ * Hence to calculate 2^10 we only need to multiply 2^8 and 2^2 skipping 2^1 and 2^4.
 */
 
-/// Function to calculate exponent in O(log(n)) using binary exponent.
+/// Recursive function to calculate exponent in O(log(n)) using binary exponent.
 int binExpo(int a, int b) {
```

Comment:
Why make your reader guess??  Why not `binary_exponent_recursive()` and `binary_exponent_iterative()`?

Reply:
Too late.

Reply:
It's clear enough up to the scope of this file (see `main()`). When one uses it else where, this is of course the best version he can get.

Reply:
I agreed with @cclauss. I think its more better to give proper justifications in function name itself. what you say? @cclauss @poyea 
So should I open a new PR or else we are going with this version?